cmake/Install.cmake

```diff
@@ -9,4 +9,107 @@
+endif()
+
+#! ly_install_directory: specifies a directory to be copied to the install layout at install time
+#
+# \arg:DIRECTORIES directories to install
+# \arg:DESTINATION (optional) destination to install the directory to (relative to CMAKE_PREFIX_PATH)
+# \arg:EXCLUDE_PATTERNS (optional) patterns to exclude
+#
+# \notes: refer to cmake's install(DIRECTORY documentation for more information
+#
+function(ly_install_directory)
+
+    set(options)
+    set(oneValueArgs DESTINATION)
+    set(multiValueArgs DIRECTORIES EXCLUDE_PATTERNS)
+
+    cmake_parse_arguments(ly_install_directory "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    if(NOT ly_install_directory_DIRECTORIES)
+        message(FATAL_ERROR "You must provide at least a directory to install")
+    endif()
+
+    foreach(directory ${ly_install_directory_DIRECTORIES})
+
+        cmake_path(ABSOLUTE_PATH directory)
+
+        if(NOT ly_install_directory_DESTINATION)
+            # maintain the same structure relative to LY_ROOT_FOLDER
+            set(ly_install_directory_DESTINATION ${directory})
+            if(${ly_install_directory_DESTINATION} STREQUAL ".")
+                set(ly_install_directory_DESTINATION ${CMAKE_CURRENT_LIST_DIR})
+            else()
+                cmake_path(ABSOLUTE_PATH ly_install_directory_DESTINATION BASE_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
+            endif()
+            # take out the last directory since install asks for the destination of the folder, without including the fodler itself
+            cmake_path(GET ly_install_directory_DESTINATION PARENT_PATH ly_install_directory_DESTINATION)
+            cmake_path(RELATIVE_PATH ly_install_directory_DESTINATION BASE_DIRECTORY ${LY_ROOT_FOLDER})       
+        endif()
+
+        unset(exclude_patterns)
+        if(ly_install_directory_EXCLUDE_PATTERNS)
+            foreach(exclude_pattern ${ly_install_directory_EXCLUDE_PATTERNS})
+                list(APPEND exclude_patterns PATTERN ${exclude_pattern} EXCLUDE)
+            endforeach()
+        endif()
+        
+        # Exclude cmake since that has to be generated
+        list(APPEND exclude_patterns PATTERN CMakeLists.txt EXCLUDE)
+        list(APPEND exclude_patterns PATTERN *.cmake EXCLUDE)
+
+        # Exclude python-related things that dont need to be installed
+        list(APPEND exclude_patterns PATTERN __pycache__ EXCLUDE)
+        list(APPEND exclude_patterns PATTERN *.egg-info EXCLUDE)     
+
+        install(DIRECTORY ${directory}
+            DESTINATION ${ly_install_directory_DESTINATION}
+            COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME} # use the deafult for the time being
+            ${exclude_patterns}
+        )
+    endforeach()
+
+endfunction()
+
+#! ly_install_files: specifies files to be copied to the install layout at install time
+#
+# \arg:FILES files to install
+# \arg:DESTINATION (optional) destination to install the directory to (relative to CMAKE_PREFIX_PATH)
```

Comment:
This comment is inaccurate. 
The DESTINATION parameter is not optional since it is causes a FATAL ERROR

Reply:
copy-paste fail